crates/ruma-common/tests/events/room_message.rs

```diff
@@ -302,6 +302,24 @@ fn markdown_content_serialization() {
     );
 }
 
+#[test]
+#[cfg(feature = "markdown")]
+fn markdown_detection() {
+    use ruma_common::events::room::message::FormattedBody;
+
+    // No markdown
+    let formatted_body = FormattedBody::markdown("A simple message.");
+    assert_matches!(formatted_body, None);
+
+    // Multiple paragraphs trigger markdown
+    let formatted_body = FormattedBody::markdown("A message\nwith\n\nmultiple\n\nparagraphs");
+    formatted_body.unwrap();
+
+    // HTML entities don't trigger markdown.
+    let formatted_body = FormattedBody::markdown("A message with & HTML < entities");
```

Comment:
This is an interesting case. When will HTML start having an effect? Is `<span>Hello</span>` considered a plain-text message?

Reply:
No, as soon as there is something that looks like a HTML tag it will be detected. At first I had written this example as `"A message with & HTML <entities>"` and it detected `<entities>` as HTML.

Reply:
Can you add a test for that?

Reply:
Done